src/materials/ConcreteASREigenstrain.C

```diff
@@ -0,0 +1,278 @@
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteASREigenstrain.h"
+
+registerMooseObject("BlackBearApp", ConcreteASREigenstrain);
+
+template <>
+InputParameters
+validParams<ConcreteASREigenstrain>()
+{
+  InputParameters params = validParams<ConcreteExpansionEigenstrainBase>();
+
+  MooseEnum expansion_type("Isotropic Anisotropic");
+  params.addParam<MooseEnum>(
+      "expansion", expansion_type, "Type of expansion resulting from volumetric strain");
+  params.addRequiredCoupledVar("temperature", "Coupled temperature");
+  params.addRequiredCoupledVar("relative_humidity", "Coupled relative humidity");
+
+  params.addParam<Real>(
+      "rh_exponent",
+      0.0,
+      "Power to which relative humidity is raised in computation of ASR volumetric strain");
+
+  params.addParam<Real>(
+      "ASR_vol_expansion",
+      0.005,
+      "Final ansymptotic ASR volumertic expansion strain when reaction is complete");
+  params.addParam<Real>(
+      "tau_c_T0", 50.0, "Chracteristic ASR time (in days) at reference temprature");
+  params.addParam<Real>("tau_L_T0", 200.0, "Latency ASR time (in days) at reference temprature");
+  params.addParam<Real>("Uc", 5400.0, "Activation energy associated with tau_c");
+  params.addParam<Real>("UL", 9400.0, "Activation energy associated with tau_L");
+  params.addParam<Real>("ASR_T0", 38.0, "Reference temperature associated ASR in Celsius");
+
+  params.addParam<Real>("f_compress", -31.0e6, "Compressive strength of concrete (in Pascals)");
+  params.addParam<Real>(
+      "alpha", 1.3333, "Constant for ASR latency time retardation under hydrostatic compression");
+  params.addParam<Real>("f_tensile", 3.0e6, "Tensile strength of concrete (in Pascals)");
+
+  params.addParam<Real>(
+      "gamma_tensile",
+      0.5,
+      "Fraction of tensile strength beyond which ASR gel is adsorbed into tensile cracks");
+  params.addParam<Real>("gamma_residual", 0.5, "Residual ASR retention factor under tension");
+
+  params.addParam<Real>(
+      "beta", 0.5, "Exponent for ASR retention factor under compressive stress state");
+
+  params.addParam<bool>("ASR_dependent_tensile_strength",
+                        false,
+                        "Set true to turn ASR reaction dependent tensile strength");
+  params.addParam<Real>(
+      "beta_f_tensile", 0.5, "Fractional residual tensile strength at full ASR reaction");
+
```

Comment:
gamma_tensile, gamma_residual and beta_f_tensile need to be between 0 and 1 (included). beta needs to be >= 0.